files/unarchive.py

```diff
@@ -579,6 +579,9 @@ def __init__(self, src, dest, file_args, module):
         self.zipflag = '-z'
         self._files_in_archive = []
 
+        # We screenscrape a huge amount of commands so use C locale anytime we do
+        module.run_command_environ_update = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C', LC_CTYPE='C')
+
```

Comment:
Even though it's more verbose, it probably makes more sense to add this as a parameter to every tar run_command.  Doing it here will lead to unexpected side effects if the code is ever changed to instantiate a TarArchive during its run but then use the ZipArchive type later (for instance, if someone changes the order that the handlers are checked in pick_archive).  See how the yum module does it if you need an example.  For instance, https://github.com/ansible/ansible-modules-core/blob/devel/packaging/os/yum.py#L281